core/controller/src/main/scala/org/apache/openwhisk/core/entitlement/PackageCollection.scala

```diff
@@ -98,7 +98,10 @@ class PackageCollection(entityStore: EntityStore)(implicit logging: Logging) ext
           val pkgOwner = namespaces.contains(binding.namespace.asString)
           val pkgDocid = binding.docid
           logging.debug(this, s"checking subject has privilege '$right' for bound package '$pkgDocid'")
-          checkPackageReadPermission(namespaces, pkgOwner, pkgDocid)
+          if (doc == pkgDocid)
```

Comment:
should this be treated as an error also?

Reply:
The way the flow works here, this is a precursor to the call into `mergePackageWithBinding`. My rationalization here was that this is merely a permissions check, and if we're checking the children of self, we've already checked self and that passed or we'd not be here. 

I'm up to suggestions on whether this should fail, however. WDYT?